diff --git a/auto/lib/openssl/conf b/nginx/auto/lib/openssl/conf
--- a/auto/lib/openssl/conf
+++ b/auto/lib/openssl/conf
@@ -65,6 +65,22 @@ else

         if [ $ngx_found = no ]; then
 
+            # CryptoPro port
+
+            ngx_feature="OpenSSL library in /opt/cprocsp/cp-openssl-1.1.0/"
+            ngx_feature_path="/opt/cprocsp/cp-openssl-1.1.0/include"
+
+            if [ $NGX_RPATH = YES ]; then
+                ngx_feature_libs="-R/opt/cprocsp/cp-openssl-1.1.0/lib/amd64 -L/opt/cprocsp/cp-openssl-1.1.0/lib/amd64 -lssl -lcrypto $NGX_LIBDL"
+            else
+                ngx_feature_libs="-L/opt/cprocsp/cp-openssl-1.1.0/lib/amd64 -lssl -lcrypto $NGX_LIBDL"
+            fi
+
+            . auto/feature
+        fi
+
+        if [ $ngx_found = no ]; then
+
             # FreeBSD port
 
             ngx_feature="OpenSSL library in /usr/local/"
